src/transport/grpc/convert.rs

```diff
@@ -0,0 +1,1347 @@
+
+        individual_data.try_into()
+    }
+}
+
+/// Implements the TryFrom trait for the AstarteDataTypeIndividual for any AstarteType.
+macro_rules! impl_astarte_type_to_individual_data_conversion_traits {
+    ($($typ:ident),*) => {
+        impl TryFrom<AstarteType> for astarte_message_hub_proto::AstarteDataTypeIndividual {
+            type Error = MessageHubProtoError;
+
+            fn try_from(d: AstarteType) -> Result<Self, Self::Error> {
+                match d {
+                    $(
+                    AstarteType::$typ(val) => Ok(val.into()),
+                    )*
+                    AstarteType::Unset => Err(MessageHubProtoError::UnsetConversion),
+                }
+            }
+        }
+    }
+}
+
+impl_astarte_type_to_individual_data_conversion_traits!(
+    Double,
+    Integer,
+    Boolean,
+    LongInteger,
+    String,
+    BinaryBlob,
+    DateTime,
+    DoubleArray,
+    IntegerArray,
+    BooleanArray,
+    LongIntegerArray,
+    StringArray,
+    BinaryBlobArray,
+    DateTimeArray
+);
+
+impl TryFrom<astarte_message_hub_proto::AstarteMessage> for AstarteDeviceDataEvent {
+    type Error = MessageHubProtoError;
+
+    fn try_from(
+        astarte_message: astarte_message_hub_proto::AstarteMessage,
+    ) -> Result<Self, Self::Error> {
+        let astarte_sdk_aggregation = match optional_chain!(astarte_message.payload)? {
+            astarte_message_hub_proto::astarte_message::Payload::AstarteData(astarte_data_type) => {
+                match optional_chain!(astarte_data_type.data)? {
+                    astarte_message_hub_proto::astarte_data_type::Data::AstarteIndividual(
+                        astarte_individual,
+                    ) => Aggregation::Individual(
+                        optional_chain!(astarte_individual.individual_data)?.try_into()?,
+                    ),
+                    astarte_message_hub_proto::astarte_data_type::Data::AstarteObject(
+                        astarte_object,
+                    ) => {
+                        let astarte_sdk_aggregation =
+                            map_values_to_astarte_type(astarte_object.object_data)?;
+                        Aggregation::Object(astarte_sdk_aggregation)
+                    }
+                }
+            }
+            astarte_message_hub_proto::astarte_message::Payload::AstarteUnset(_) => {
+                Aggregation::Individual(AstarteType::Unset)
+            }
+        };
+
+        Ok(AstarteDeviceDataEvent {
+            interface: astarte_message.interface_name,
+            path: astarte_message.path,
+            data: astarte_sdk_aggregation,
+        })
+    }
+}
+
+impl TryFrom<astarte_message_hub_proto::types::InterfaceJson> for crate::Interface {
+    type Error = crate::Error;
+
+    fn try_from(
+        interface: astarte_message_hub_proto::types::InterfaceJson,
+    ) -> Result<Self, Self::Error> {
+        let interface_str = String::from_utf8_lossy(&interface.0);
```

Comment:
This should be an hard error, you can use `str::from_utf8`

Reply:
That's correct.